src/FixedPointDecimals.jl

```diff
@@ -38,6 +38,37 @@ import Base: reinterpret, zero, one, abs, sign, ==, <, <=, +, -, /, *, div, rem,
 
 include("fldmod_by_const.jl")
 
+# =========================================================
+# Set Int128 to widen to Int256 within FixedPointDecimals to avoid allocations from BigInt.
+# ---------------------------------------
+
+import BitIntegers
+
+# ---------- Necessary Extensions -----------------------
+# TODO: delete these once https://github.com/rfourquet/BitIntegers.jl/pull/2 and
+# https://github.com/rfourquet/BitIntegers.jl/pull/3 are merged.
+
+# BitIntegers is missing unsigned(::Type{Int256})
+# XBI == Union{BitIntegers.AbstractBitSigned, BitIntegers.AbstractBitUnsigned}
+Base.unsigned(::Type{T}) where T<:BitIntegers.XBI = typeof(convert(Unsigned, zero(T)))
+Base.unsigned(x::T) where T<:BitIntegers.XBI = reinterpret(unsigned(T), x)
+
+# BitIntegers is missing iseven/isodd
+# Prevent expensive calculation for Int256. Needed since `_round_to_even` calls `iseven`.
+Base.isodd(a::BitIntegers.Int256) = Base.isodd(a % Int)  # only depends on the final bit! :)
```

Comment:
These are monkey patching functionality that belongs in a different package (neither `BitIntegers` nor `Base.isodd` are "owned" by this package). Sometimes, this can lead to method overwriting warnings when the downstream packages are updated (although I suspect it won't in this case, but I'd prefer to avoid it overall). Would it be reasonable to simply wait for `BitIntegers` to merge the appropriate PRs?

Reply:
:) Yeah i think that's reasonable! I agree.
I'm linking those two PRs here for posterity:
https://github.com/rfourquet/BitIntegers.jl/pull/3
https://github.com/rfourquet/BitIntegers.jl/pull/2

The `unsigned(::Type{T})` one, https://github.com/rfourquet/BitIntegers.jl/pull/2, might take a while because I think it's waiting for some progress on https://github.com/JuliaLang/julia/issues/30444.